src/main/java/com/dataartisans/flink/dataflow/translation/wrappers/streaming/FlinkGroupAlsoByWindowWrapper.java

```diff
@@ -491,6 +516,9 @@ public BoundedWindow window() {
 
 				@Override
 				public void outputWindowedValue(KV<K, VOUT> output, Instant timestamp, Collection<? extends BoundedWindow> windows, PaneInfo pane) {
+					// TODO: No need to represent timestamp twice.
+					// collector.setAbsoluteTimestamp(timestamp.getMillis());
+					collector.setAbsoluteTimestamp(0);
 					collector.collect(WindowedValue.of(output, timestamp, windows, pane));
```

Comment:
I think going further, and using some kind of WindowedValueWithoutTimestamp would be even better. The only extra value we need here is the window set.